tests/e2e/testsLibrary/LsTests.ts

```diff
@@ -8,81 +8,88 @@
  * SPDX-License-Identifier: EPL-2.0
  **********************************************************************/
 
-import { e2eContainer } from '../inversify.config';
-import { Editor, CLASSES, Ide, TimeoutConstants } from '..';
+import 'reflect-metadata';
+import { inject, injectable } from 'inversify';
+import { CLASSES } from '../inversify.types';
+import { TimeoutConstants } from '../TimeoutConstants';
+import { Editor } from '../pageobjects/ide/Editor';
+import { Ide} from '../pageobjects/ide/Ide';
 import { Key, error } from 'selenium-webdriver';
 import { Logger } from '../utils/Logger';
 
-const editor: Editor = e2eContainer.get(CLASSES.Editor);
-const ide: Ide = e2eContainer.get(CLASSES.Ide);
+@injectable()
+export class LsTests {
 
- export function errorHighlighting(openedTab: string, textToWrite: string, line: number) {
-    test('Error highlighting', async () => {
-        await editor.type(openedTab, textToWrite, line);
-        try {
-            await editor.waitErrorInLine(line);
-        } catch (err) {
-            if (!(err instanceof error.TimeoutError)) {
-                throw err;
+    constructor(@inject(CLASSES.Editor) private readonly editor: Editor, @inject(CLASSES.Ide) private readonly ide: Ide) { }
+
+    public errorHighlighting(openedTab: string, textToWrite: string, line: number) {
+        test('Error highlighting', async () => {
+            await this.editor.type(openedTab, textToWrite, line);
+            try {
+                await this.editor.waitErrorInLine(line);
+            } catch (err) {
+                if (!(err instanceof error.TimeoutError)) {
+                    throw err;
+                }
+                // try again
+                Logger.debug(`Waiting for squiggly-error failed for the first time (issue #16166)`);
+                await this.editor.waitErrorInLine(line);
+            }
+            for (let i = 0; i < textToWrite.length; i++) {
```

Comment:
A bit hard to read method, what about a bit separate by whitespaces different steps, and for example it would be great to add comment to this loop or even better, move it to the specific method. Let me clarify, when you read steps of the method
you see understandable sequence:
 - typeText
 - waitError
 - <code> (hm, what )
 but when you face with a code you need to investigate its steps to understand what happens here.
  - typeText
 - waitError
 - <code> (what is happening here)

Reply:
ok, I'll try to simplify it

Reply:
I will have to do it in a separate PR, I'll open an issue for this task, it's out of scope of this issue and MR and this merge is blocking me from tasks I need to do on CRW

Reply:
Will be fixed as part of https://github.com/eclipse/che/issues/19749